wvd-sh/WVD scaling script/basicScale-MSI.ps1

```diff
@@ -742,19 +780,26 @@ else {
                 foreach ($sessionHost in $RDSessionHost.sessionhostname) {
                     if ($hostPoolUserSessions.Count -ge $AvailableSessionCapacity) {
                         $VMName = $sessionHost.Split(".")[0]
+                          $VmInfo = Get-AzureRmVM | Where-Object { $_.Name -eq $VMName }
+                            #Check the Session host is in Maintainance
+                            if($VmInfo.Tags.Keys -contains $maintainanceTagName){
+                                 Write-Log 1 "Session Host is in Maintainace: $($sessionhost | out-string)"
+                                 Continue
+                                }
+
                         $roleInstance = Get-AzureRmVM -Status | Where-Object { $_.Name.Contains($VMName) }
 
                         if ($sessionHost.ToLower().Contains($roleInstance.Name.ToLower())) {
-                            #check if the Azure VM is running or not
-
-                            if ($roleInstance.PowerState -ne "VM running") {
-                                $getShsinfo = Get-RdsSessionHost -TenantName $tenantname -HostPoolName $hostPoolName
+                            #check if the Azure VM is running or not and session host is healthy or not
+                            $getShsinfo = Get-RdsSessionHost -TenantName $tenantname -HostPoolName $hostPoolName -Name $sessionHost
+                            if ($roleInstance.PowerState -ne "VM running" -and $getShsinfo.UpdateState -eq "Succeeded") {
+                                #Check the session host is allowing new sessions or not
```

Comment:
"check if the session host is allowing new connections"

Reply:
Thank you Christian, we have updated accordingly